google/resource_container_cluster.go

```diff
@@ -315,6 +315,106 @@ func resourceContainerCluster() *schema.Resource {
 										Default:     "default",
 										Description: `The Google Cloud Platform Service Account to be used by the node VMs.`,
 									},
+									"shielded_instance_config": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										Description: `Shielded Instance options.`,
+										MaxItems:    1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"enable_secure_boot": {
+													Type:        schema.TypeBool,
+													Optional:    true,
+													Default:     false,
+													Description: `Defines whether the instance has Secure Boot enabled.`,
+												},
+												"enable_integrity_monitoring": {
+													Type:        schema.TypeBool,
+													Optional:    true,
+													Default:     true,
+													Description: `Defines whether the instance has integrity monitoring enabled.`,
+												},
+											},
+										},
+									},
+									"upgrade_settings": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										MaxItems:    1,
+										Description: `Specifies the upgrade settings for NAP created node pools.`,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"max_surge": {
+													Type:         schema.TypeInt,
+													Required:     true,
+													ValidateFunc: validation.IntAtLeast(0),
+													Description:  `The number of additional nodes that can be added to the node pool during an upgrade. Increasing max_surge raises the number of nodes that can be upgraded simultaneously. Can be set to 0 or greater.`,
+												},
+
+												"max_unavailable": {
+													Type:         schema.TypeInt,
+													Required:     true,
+													ValidateFunc: validation.IntAtLeast(0),
+													Description:  `The number of nodes that can be simultaneously unavailable during an upgrade. Increasing max_unavailable raises the number of nodes that can be upgraded in parallel. Can be set to 0 or greater.`,
+												},
+											},
+										},
+									},
+									"management": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										MaxItems:    1,
+										Description: `Specifies the node management options for NAP created node-pools.`,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"auto_repair": {
+													Type:        schema.TypeBool,
+													Optional:    true,
+													Default:     false,
+													Description: `Whether the nodes will be automatically repaired.`,
+												},
+
+												"auto_upgrade": {
+													Type:        schema.TypeBool,
+													Optional:    true,
+													Default:     false,
+													Description: `Whether the nodes will be automatically upgraded.`,
+												},
+											},
+										},
+									},
+									"min_cpu_platform": {
```

Comment:
min_cpu_platform already exists in the beta provider. We can add it by removing the version guard tags in the upstream repo here and in the flatten/expand methods: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/third_party/terraform/resources/resource_container_cluster.go.erb#L442

Similarly, the other fields can be added to that file to add to both the google and google-beta providers.

We will also need these fields to be documented in the container_cluster markdown file, and test the update behavior for these fields by changing their values in the test.

Reply:
Hey, thanks for the indications, I gave it a try in https://github.com/GoogleCloudPlatform/magic-modules/pull/5490 .